### PR TITLE
Added `PUBLIC_NATIVE_HEADERS` to the `FRAMEWORK_HEADERS` list in `jre_core_framework.mk`

### LINKED ISSUE #2170: JRE_Core.xcframework is missing headers
I added `JRE_Core.xcframework` to a project. When I import any of its headers (e.g. `<java/lang/Integer.h>`) into the Objective-C files in my project , I get a `'J2ObjC_header.h' file not found` build error.

On comparing the headers in `JRE_Core.xcframework` against the headers in `JRE.xcframework`, I see that `JRE.xcframework` contains `J2ObjC_header.h` as well as a number of other headers that probably should be in `JRE_Core.xcframework`.

The headers which are missing from `JRE_Core.xcframework` seem to be the ones contained in the `PUBLIC_NATIVE_HEADERS` list in the [jre_emul/jre_sources.mk][1] file.

In order to add the missing headers to `JRE_Core.xcframework`, I am thinking to change the `FRAMEWORK_HEADERS` declaration in [jre_emul/subset_frameworks/jre_core_framework.mk][2] to the following:

```none
FRAMEWORK_HEADERS = $(PUBLIC_NATIVE_HEADERS) $(JAVA_PUBLIC_SOURCES_CORE:.java=.h)
```

Is this the right thing to do?

[1]: https://github.com/google/j2objc/blob/master/jre_emul/jre_sources.mk
[2]: https://github.com/google/j2objc/blob/master/jre_emul/subset_frameworks/jre_core_framework.mk

### PR DESCRIPTION
**Pull request by adil-hussain-84**
Added `PUBLIC_NATIVE_HEADERS` to the `FRAMEWORK_HEADERS` list in `jre_emul/subset_frameworks/jre_core_framework.mk` so that the `PUBLIC_NATIVE_HEADERS` are added to the `JRE_Core.xcframework` folder which is created by the mk file. Fixes #2170.

### CODE REVIEW

**Review by tomball — APPROVED** [review]
LGTM

**Comment by adil-hussain-84** [comment]
@tomball: Thanks for the approval 👍 Can we get this merged in please 🙏

**Comment by tomball** [comment]
Argh! There's a new automation for accepting PRs that works better but now
requires two approvers. I've asked the other j2objc engineers to approve,
so whoever gets to it first will unblock it.

**Pull request merged**
